Modix.Services/ErrorHandling/DiscordResultFormatter.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using MediatR;
+using Modix.Common.ErrorHandling;
+using Modix.Services.Core;
+
+namespace Modix.Services.ErrorHandling
+{
+    /// <summary>
+    /// Formats instances of <typeparamref name="TResult"/> into <see cref="EmbedBuilder"/>s for sending to Discord
+    /// </summary>
+    public abstract class DiscordResultFormatter<TInput> : IResultFormatter<TInput, EmbedBuilder> where TInput : ServiceResult
+    {
+        protected ICommandContext Context { get; private set; }
+
+        public DiscordResultFormatter(ICommandContextAccessor contextAccessor)
+        {
+            Context = contextAccessor.Context;
+        }
+
+        /// <inheritdoc />
+        public virtual EmbedBuilder Format(TInput result)
+        {
+            if (result.IsSuccess) { return null; }
+
+            return new EmbedBuilder()
+                .WithTitle("Uh oh, an error occurred.")
+                .WithDescription(result.Error)
+                .WithColor(new Color(255, 0, 0));
+        }
+    }
+
+    //We need to seal the generic otherwise DI is not happy with us
+    /// <inheritdoc />
+    public class DefaultDiscordResultFormatter : DiscordResultFormatter<ServiceResult>
```

Comment:
I'd say the `Format` method above should be abstract, and you move the contents down here. We don't expect any `DiscordResultFormatter` subclasses to NOT override it, right? Any subclass that leaves it as-is could just be deleted entirely and the default formatter would take its place.

Reply:
Good point - refactored.